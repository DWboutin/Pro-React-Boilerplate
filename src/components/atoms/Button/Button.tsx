import { FunctionComponent, ReactNode } from 'react'
import styled from 'styled-components'

const Container = styled.div`
  padding: 10px;
  background-color: blue;
`

export interface Props {
  children: ReactNode
}

const Button: FunctionComponent<Props> = ({ children }) => {
  return <Container>{children}</Container>
}

export default Button
