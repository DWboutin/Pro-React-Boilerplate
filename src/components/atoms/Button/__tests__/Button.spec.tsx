import renderer from 'react-test-renderer'

import Button from '../Button'

describe('Button', () => {
  it('should renders correctly', () => {
    const tree = renderer.create(<Button>Button text</Button>).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
