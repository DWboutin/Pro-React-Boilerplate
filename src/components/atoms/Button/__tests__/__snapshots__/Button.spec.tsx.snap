// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders correctly 1`] = `
<div
  className="sc-bcXHqe hGKNqF"
>
  Button text
</div>
`;
